src/mgr/MetricTypes.h

```diff
@@ -30,6 +30,23 @@ struct OSDMetricPayload {
     denc(v.report, p);
     DENC_FINISH(p);
   }
+
+  void dump(ceph::Formatter *f) const {
+    f->open_array_section("report");
+    for (auto& i : report) {
```

Comment:
C++17's structured bindings could useful to provide names for the `std::pair`'s members (`query` & `report`?).